bc_obps/compliance/tests/api/test_permissions.py

```diff
@@ -0,0 +1,130 @@
+from unittest.mock import MagicMock
+from model_bakery.baker import make_recipe
+import pytest
+from compliance.api.permissions import (
+    check_compliance_version_ownership_in_url,
+)
+
+
+pytestmark = pytest.mark.django_db
+
+role_status_access_tuples = [
+    ("pending", "Approved", False),
+    ("pending", "Pending", False),
+    ("pending", "Declined", False),
+    ("admin", "Pending", False),
+    ("admin", "Declined", False),
+    ("reporter", "Pending", False),
+    ("reporter", "Pending", False),
+    ("admin", "Approved", True),
+    ("reporter", "Approved", True),
+]
+
+
+class TestVersionOwnershipFromUrl:
+    @pytest.mark.parametrize("role,status,expected_validity", role_status_access_tuples)
+    def test_compliance_version_ownership_from_url(self, role, status, expected_validity):
+        # Setup
+        user_operator = make_recipe("registration.tests.utils.user_operator", role=role, status=status)
+
+        compliance_report_version = make_recipe(
+            "compliance.tests.utils.compliance_report_version",
+            compliance_report__report__operator=user_operator.operator,
+            compliance_report__created_at='2024-06-15',
+            # compliance_report__report__operation=operation,
+        )
+
+        validator_under_test = check_compliance_version_ownership_in_url("test_id")
+
+        mock_request = MagicMock()
+        mock_request.resolver_match.kwargs = {"test_id": compliance_report_version.id}
+        mock_request.current_user = user_operator.user
+
+        is_valid = validator_under_test(mock_request)
+
+        assert is_valid == expected_validity
+
+    def test_version_ownership_from_url_denies_access_if_no_timeline_record(self):
+        compliance_report_version = make_recipe("compliance.tests.utils.compliance_report_version")
+        user = make_recipe("registration.tests.utils.industry_operator_user")
+
+        validator_under_test = check_compliance_version_ownership_in_url("test_id")
+
+        mock_request = MagicMock()
+        mock_request.resolver_match.kwargs = {"test_id": compliance_report_version.id}
+        mock_request.current_user = user
+
+        is_valid = validator_under_test(mock_request)
+
+        assert not is_valid
+
+    def test_version_ownership_from_url_denies_access_if_operation_transferred(self):
```

Comment:
Is this test still necessary? We only check direct relationships, not those through the timeline table.